pkg/forman/forman_test.go

```diff
@@ -0,0 +1,119 @@
+package forman
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+)
+
+func TestGoForman(t *testing.T) {
+	workerCount := 2
+	// totalRequest is the total number of requests to be processed to delay work
+	totalRequest := 10
+	// Create a counter to track the number of requests processed
+	requestCounter := sync.Map{}
+	workWithDelay := "test-work-with-delay"
+	requeueWork := "test-work-requeue"
+	panicWork := "test-panic"
+
+	// Create a callback function to track the processing of requests
+	callback := func(req Request) {
+		val, ok := requestCounter.Load(req.Name)
+		if ok {
+			requestCounter.Store(req.Name, val.(int)+1)
+			return
+		}
+		requestCounter.Store(req.Name, 0)
```

Comment:
- small req, can the condition & statement be reversed?
- let's have a condition for unlikely branch in long run first
```suggestion
		if !ok {
			requestCounter.Store(req.Name, 0)
		}
		requestCounter.Store(req.Name, val.(int)+1)
```

Reply:
Yes it can be done but its missing a branching, it's missing an else statement, or a return is required inside the branch.

Reply:
- sry, I didn't get that. Correct me if I understood incorrectly,
1. if key doesn't exist, initialize value
2. if key exists, increment value
- so my suggestion was, if key doesn't exist, increment value by initializing the value first

Reply:
Yes that is correct, in the above `if` check there is a return missing. it should not initialize and increment at the same time.

Reply:
> it should not initialize and increment at the same time.
- Ya, isn't that a issue? Counting initialization also an access and as a result asserting the test not for equality but for greater count at the end of the file?
- Pardon dragging this, I felt that assertions at the end unusual and thought the callback might be the reason.

Reply:
IMO that is not the usual flow, either we initialize or increment not both at the same time, if you still feel it need to be changed I can change it and also change the assert as per it.

Reply:
sry, I don't want this to be changed just like that. If you are want to consider initialization as a count, ntg to change here.